Docker/Container.php

```diff
@@ -336,20 +332,30 @@ private function handleContainerFailure(Process $process, $startTime)
         // put the whole message to exception data, but make sure not use too much memory
         $data = [
             "output" => mb_substr($output, -1000000),
-            "errorOutput" => mb_substr($errorOutput, -1000000)
+            "errorOutput" => mb_substr($errorOutput, -1000000),
+            "container" => [
+                "id" => $this->getId(),
+                "image" => $this->getImage()->getFullImageId(),
+            ],
         ];
 
-        if ($process->getExitCode() == 1) {
-            $data["container"] = [
-                "id" => $this->getId(),
-                "image" => $this->getImage()->getFullImageId()
-            ];
-            // syrup will log the process error output as part of the exception body trimmed, so log the entire (if possible) error as a separate event
-            $buffer = $process->getErrorOutput();
-            if (mb_strlen($buffer) > 64000) {
-                $buffer = mb_substr($buffer, 0, 64000) . " [trimmed]";
+        // killed containers
+        if ($process->getExitCode() == 137) {
+            // this catches the timeout from `sudo timeout`
+            if ($duration >= $this->getImage()->getSourceComponent()->getProcessTimeout()) {
+                throw new UserException(
+                    "Running {$this->getImage()->getFullImageId()} container exceeded the timeout of " .
+                    $this->getImage()->getSourceComponent()->getProcessTimeout() . " seconds.",
+                    null,
+                    $data
+                );
+            } else {
+                throw new InitializationException(
+                    "{$this->getImage()->getFullImageId()} container terminated. Will restart."
+                );
             }
-            $this->containerLogger->error($buffer);
+        } elseif ($process->getExitCode() == 1) {
+            // syrup will log the process error output as part of the exception body
```

Comment:
Takže v případě user error to teda zařídí syrup že se to hodí do resultu jobu + zaloguje event s chybou?

Reply:
zaloguje https://github.com/keboola/syrup/blob/master/src/Keboola/Syrup/Command/JobCommand.php#L223 a vrati message v resultu jobu a skonci user chybou https://github.com/keboola/syrup/blob/master/src/Keboola/Syrup/Command/JobCommand.php#L230

Reply:
j jasny v tomhle jsem se zrovna vrtal a tady se rozlisi jestli tam message pujde ven pro user error nebo se schova pro internal https://github.com/keboola/syrup/blob/master/src/Keboola/Syrup/Monolog/Handler/StorageApiHandler.php#L92 takze ok